### PR TITLE
exim: attempt to fix build on 10.11 and earlier

### PR DESCRIPTION
**Pull request by chrstphrchvz**
#### Description
See build failures: https://ports.macports.org/port/exim/builds

###### ~~Tested on~~

Untested.

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: mail/exim/Portfile [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 PortSystem              1.0
 
+# clock_gettime(), openat()
+PortGroup               legacysupport 1.1
+legacysupport.newest_darwin_requires_legacy 15
```
Comment (cjones051073) [inline mail/exim/Portfile:7]: Please add an empty line after the port group to silence the lint warnings

**Review by cjones051073 — CHANGES_REQUESTED** [review]

**Review by cjones051073 — APPROVED** [review]

**Comment by chrstphrchvz** [comment]
Fails because the `-lMacportsLegacySupport` flag isn’t passed to the linker.

I notice that the `clock_gettime()` usage seems optional, so maybe the check being used (`#ifdef _POSIX_MONOTONIC_CLOCK`) does not work properly.

**Comment by cjones051073** [comment]
no, the check is fine as it just tests for the declaration in the headers, which will be found. The issue is the build system is not properly passing the linker flags, and if you do one and not the other you have problems...

**Comment by chrstphrchvz** [comment]
> no, the check is fine as it just tests for the declaration in the headers, which will be found.

I meant without legacy support—that the check is not useful on macOS because its headers define `_POSIX_MONOTONIC_CLOCK` even for macOS 10.11 and earlier which lack `clock_gettime`, i.e. an upstream bug (if they intend to support older macOS).

**Comment by chrstphrchvz** [comment]
> i.e. an upstream bug (if they intend to support older macOS).

…unless one argues this is a macOS bug (POSIX noncompliance)

**Pull request merged**